components/layout/display_list/builder.rs

```diff
@@ -2828,30 +2826,24 @@ impl BaseFlow {
     }
 }
 
-trait ComputedValuesCursorUtility {
-    fn get_cursor(&self, default_cursor: CursorKind) -> Option<CursorKind>;
-}
-
-impl ComputedValuesCursorUtility for ComputedValues {
-    /// Gets the cursor to use given the specific ComputedValues.  `default_cursor` specifies
-    /// the cursor to use if `cursor` is `auto`. Typically, this will be `PointerCursor`, but for
-    /// text display items it may be `TextCursor` or `VerticalTextCursor`.
-    #[inline]
-    fn get_cursor(&self, default_cursor: CursorKind) -> Option<CursorKind> {
-        match (
-            self.get_inherited_ui().pointer_events,
-            &self.get_inherited_ui().cursor,
-        ) {
-            (PointerEvents::None, _) => None,
-            (
-                PointerEvents::Auto,
-                &Cursor {
-                    keyword: CursorKind::Auto,
-                    ..
-                },
-            ) => Some(default_cursor),
-            (PointerEvents::Auto, &Cursor { keyword, .. }) => Some(keyword),
-        }
+/// Gets the cursor to use given the specific ComputedValues.  `default_cursor` specifies
+/// the cursor to use if `cursor` is `auto`. Typically, this will be `PointerCursor`, but for
+/// text display items it may be `TextCursor` or `VerticalTextCursor`.
+#[inline]
+fn get_cursor(values: &ComputedValues, default_cursor: CursorKind) -> Option<CursorKind> {
+    match (
```

Comment:
You could also just move it to style, your call.

In any case I think this is simpler to read as follows, if you want to make that change:

```rust
let ui = values.get_inherited_ui();
if ui.pointer_events == PointerEvents::None {
    return None;
}

match ui.cursor {
    // ...
}
```

Reply:
Made some changes like you suggested but didn't move it to style given there is no other user.